src/test/java/edu/ksu/canvas/attendance/submitter/AssignmentValidatorUTest.java

```diff
@@ -0,0 +1,148 @@
+
+    private AssignmentValidator assignmentValidator;
+
+    @Mock
+    private CanvasApiWrapperService canvasApiWrapperService;
+
+    private NonRefreshableOauthToken oauthToken;
+    private AttendanceAssignment attendanceAssignment;
+    private Optional<Assignment> assignmentOptional;
+    private Assignment assignment;
+    private AttendanceAssignment assignmentConfigurationFromSetup;
+    private AttendanceSection attendanceSection;
+
+    @Before
+    public void setup() {
+
+        assignmentValidator = new AssignmentValidator();
+
+        oauthToken = new NonRefreshableOauthToken(OAUTH_STRING);
+
+        assignmentConfigurationFromSetup = new AttendanceAssignment();
+        assignmentConfigurationFromSetup.setAbsentPoints(0.0);
+        assignmentConfigurationFromSetup.setAssignmentPoints(ASSIGNMENT_POINTS);
+        assignmentConfigurationFromSetup.setExcusedPoints(0.0);
+        assignmentConfigurationFromSetup.setPresentPoints(100.0);
+        assignmentConfigurationFromSetup.setTardyPoints(0.0);
+
+        attendanceSection = new AttendanceSection();
+        attendanceSection.setName(SECTION_NAME);
+        attendanceSection.setCanvasCourseId(CANVAS_COURSE_ID);
+        attendanceSection.setSectionId(SECTION_ID);
+        attendanceSection.setCanvasSectionId(CANVAS_SECTION_ID);
+
+        attendanceAssignment = new AttendanceAssignment();
+        attendanceAssignment.setAssignmentId(ASSIGNMENT_ID);
+        attendanceAssignment.setAssignmentName(ASSIGNMENT_NAME);
+        attendanceAssignment.setCanvasAssignmentId(CANVAS_ASSIGNMENT_ID);
+        attendanceAssignment.setGradingOn(true);
+        attendanceAssignment.setAssignmentPoints(ASSIGNMENT_POINTS);
+        attendanceAssignment.setAbsentPoints(0.0);
+        attendanceAssignment.setExcusedPoints(0.0);
+        attendanceAssignment.setTardyPoints(0.0);
+        attendanceAssignment.setPresentPoints(100.0);
+        attendanceAssignment.setAttendanceSection(attendanceSection);
+        attendanceAssignment.setStatus(AttendanceAssignment.Status.UNKNOWN);
+
+
+        assignment = new Assignment();
+        assignment.setPointsPossible(ASSIGNMENT_POINTS);
+        assignmentOptional = Optional.of(assignment);
+
+    }
+
+    @Test
+    public void validateAttendanceAssignmentHappyPath() throws IOException, AttendanceAssignmentException {
+        when(canvasApiWrapperService.getSingleAssignment(COURSE_ID, oauthToken, CANVAS_ASSIGNMENT_ID+"")).thenReturn(assignmentOptional);
+        assignmentValidator.validateAttendanceAssignment(COURSE_ID, attendanceAssignment, canvasApiWrapperService, oauthToken);
+        Assert.assertNotEquals(AttendanceAssignment.Status.NOT_LINKED_TO_CANVAS, attendanceAssignment.getStatus());
+        Assert.assertNotEquals(AttendanceAssignment.Status.CANVAS_AND_DB_DISCREPANCY, attendanceAssignment.getStatus());
+    }
+
+    @Test
+    public void validateCanvasAssignmentHappyPath() throws IOException, AttendanceAssignmentException {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Refactor this method to throw at most one checked exception instead of: java.io.IOException, edu.ksu.canvas.attendance.exception.AttendanceAssignmentException [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.ome.ksu.edu/coding_rules#rule_key=squid%3AS1160)